main.cpp

```diff
@@ -5,28 +5,112 @@ int main_dialog::idd() const {
 	return IDD_DIALOG;
 }
 bool main_dialog::on_init_dialog() {
+	set_text(IDC_EDIT1, tstr);
 	return true;
 }
 bool main_dialog::on_ok() {
+	tstr = get_text(IDC_EDIT1);
 	return true;
 }
 
+void get_font(HWND parent, LOGFONT& lf, COLORREF& cr)
+{
+	CHOOSEFONT cf{ sizeof CHOOSEFONT };
+	cf.Flags = CF_INITTOLOGFONTSTRUCT | CF_SCREENFONTS | CF_EFFECTS;
+	cf.hwndOwner = parent;
+	cf.lpLogFont = &lf;
+	::ChooseFont(&cf);
+	cr = cf.rgbColors;
+}
+
+
+COLORREF get_color(HWND parent, COLORREF cur) {
+	COLORREF custom_colors[16]{ 0 };

```

Comment:
Ovo ne pomaže ako nije _static_ (ili global) - svaki puta će biti 16 crnih boja (umjesto zapamćenih "Add to custom colors")